CryptoPricesKIFUITests/Sources/Specs/HomeSpec.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  HomeSpec.swift
+//  CryptoPricesKIFUITests
+//
+//  Created by Bliss on 11/1/23.
+//
+
+@testable import Home
+import Nimble
+import Quick
+
+final class HomeSpec: QuickSpec {
```

Comment:
It looks like we are putting this KIF test in the main project, but since we are using a modular architecture, this UI test should be placed directly in the `Home` package instead. What do you guys think? @Thieurom @nkhanh44

Reply:
Firstly, welcome to CryptoPrices @blyscuit 🎉 

@minhnimble Yes, that's exactly what I was going to say 😄 

For our architecture, the tests of a module will be placed in the corresponding package. So basically we would have two new test targets for the `Home` and `MyCoin` respectively.

The `CryptoPricesKIFUITests` target you created is still useful if we want to make some kind of _integration test_, like tapping a row on the TrendingCoins section on the Home will navigate to the MyCoin with correct UIs, ...

I think today I'm gonna take a look at [KIF](https://github.com/kif-framework/KIF) to check how to best integrate it into our project 🙏

Reply:
A warm welcome to @blyscuit 🕺 .
@minhnimble @Thieurom I agree 💪 . I used to use KIF before but in our architecture, it's something I need to research a little bit :D. I will take a look at this after my tasks are done 🙏 .

Reply:
KIF is "an iOS integration test framework", it requires some sort of 'app' to run the test. Underneath it works similarly to XCUITest (meant as a replacement of XCUITest): Find view in app > interact with app via accessibility components.

I think this discussion is referring to Unit test of the views, which can be done through [ViewInspector](https://github.com/nalexn/ViewInspector) that David presented or with [Snapshot Testing](https://github.com/pointfreeco/swift-snapshot-testing) that Thieu briefly discussed.

Reply:
The reason I'm pushing `KIF` to the chapter and this project is because iOS dev are less likely to want to add integration test because how fragile `XCUITest` is. But with `KIF` we can be confident about our integration test and is way easier to work with.

Reply:
On the modular architecture note, I think it will make sense to add `KIF` to the module if the module act as a user's story ie: register, kyc, feature1, feature2. Currently this modular architecture is separating the feature module to smaller UI modules.

Reply:
Thank you for your insights @blyscuit 🙏

> KIF is "[an iOS integration test framework](https://github.com/kif-framework/KIF#kif-ios-integration-testing-framework)", it requires some sort of 'app' to run the test.

That's what I thought in the first place
> The CryptoPricesKIFUITests target you created is still useful if we want to make some kind of integration test, like tapping a row on the TrendingCoins section on the Home will navigate to the MyCoin with correct UIs, ...

So basically, with the same `KIF` framework (without an additional one like `ViewInspector`) we can achieve 2 purposes:
- Add the _Unit test of the views_ for each module. It needs some extra work like wrap in a `UIHostingController` (it can be extracted to a helper/extension though) among other things. I think we don't have time for this now!
- Add the _integration tests_, which need to be implemented on the app level like you did here. From what I've seen, it's quite similar with another tool like Appium (I remembered my QA guys and gals wrote alot of test scripts, scenarios in my former companies with this 😄). We should keep going with this! Still I need sometime to play around with `KIF` before making some feedback 🙏

Reply:
Your two points here are correct.

> It needs some extra work like wrapping in a `UIHostingController`

With the current structure of the modules (`Home` and `MyCoin`), the UIHostingController would also be implemented in a new Target (same or new project file) because the Swift module has limitation of creating an app target inside the module.

> it's quite similar to another tool like Appium

Same as Appium, but much faster build & run and written in Obj-c/Swift.